sklearn/manifold/tests/test_t_sne.py

```diff
@@ -276,7 +276,7 @@ def test_fit_csr_matrix():
     X[(np.random.randint(0, 100, 50), np.random.randint(0, 2, 50))] = 0.0
     X_csr = sp.csr_matrix(X)
     tsne = TSNE(n_components=2, perplexity=10, learning_rate=100.0,
-                random_state=0, method='exact')
+                random_state=0, method='exact', metric="euclidean")
```

Comment:
You should not need this change if the default behavior is unchanged.

Reply:
Default behavior _is_ changed: It is now `metric='minkowski'`. I understand that in particular case of `p=2`, it is equivalent to euclidean distance. However, this test case is written to check sparse matrix functionality (which scipy distance metrics do not support). Thus, it is necessary to explicitly mention `metric='euclidean'` so it does not use scipy distance metrics.

Reply:
So running `TSNE().fit(X_csr)` works in master but not with the metric change in this PR, doesn't it?
We should not break user code this way.